### DISCUSSION TITLE
I want to help with `report` flag

### DISCUSSION THREAD

**Discussion by lnxfsf** [elrumo/macOS_Big_Sur_icons_replacements#1462] [Help with the website]
Hi, 
I want to help
Need some guidance. 

<br><br>

@Zabriskije mentioned 
- https://github.com/elrumo/macOS_Big_Sur_icons_replacements/issues/1456  

which touches on idea from #1413. 
So @Zabriskije proposed solution with design for individual card he have in mind how it should have. 

<br><br><br>

So what we need is:

- share links to specific icons. So it means every icon have it's link which opens that icon page. But instead of every icon having it's separate page, I think we should just highlight that icon. That icon to come first, like when we search, and only one icon appears. Which is that one. Highlighted. 

- And we should put share icon somewhere here on hover, possibly on bottom right side ?  
<img src="https://user-images.githubusercontent.com/101254295/237693946-e28e6a6c-2fbf-4940-8263-a2d8195dd438.png">

- Based on image. On hover to have "flag" icon to report for copyrighted material. 
But then, to who it will be reported to ? Who will receive it, to be able to take it down ? 

- And then @Zabriskije what this `info` button means ? When it shows up ? When it's not hovered ? But when icon is hovered how I can click on that `info` button to show those details  ? Or you meant that `info` button (one in the middle) is also part of hover with heart and report icon, just that `info` icon is clickable ?  

- On upload of pictures, it means we need to ask for "Author licence link" ? What if user can't find one ? And what it would mean in this context ? 

<br><br>

-  We have downloads counter
-  We have favorite (heart) button

**Comment by Zabriskije** [comment]
> And then @Zabriskije what this info button means ? When it shows up ? When it's not hovered ? But when icon is hovered how I can click on that info button to show those details ? Or you meant that info button (one in the middle) is also part of hover with heart and report icon, just that info icon is clickable ?

I meant to make them appear on hover as the favorite does (your last option). I didn't include the other ones in the second/third image to focus on the info button/panel.

> On upload of pictures, it means we need to ask for "Author licence link" ? What if user can't find one ? And what it would mean in this context ?

It was just an example of what to include in the info panel. I think It shouldn't be mandatory but optional. Only the link to the author's work will be mandatory if the "I'm the author of this icon" checkbox isn't selected.

---

This is a complicated feature that will slow down icon approval if only one or two people are involved in content moderation. This also means that the moderators should have a way to interact with the uploader to discuss possible problems such as broken links, non-explicit sharing license on the author's page, etc.; which will request some internal tool for issue tracking (assuming there's none right now), and all the hassle that comes with it.

Pondering about this again, I think it's just better to not allow others' work and only add the "Report" flag. It's odd to see a well-known icon designed by a well-known designer shared under a random user profile. The current "I'm the author of this icon" should have mitigated this problem but it's not reinforced. I would suggest concentrating efforts on better scrutiny/takedowns instead of this burden of code and manpower.

The only thing left out will be derivative work, which can be displayed in the aftermentioned info panel, or under the uploader username, like this:

<img width="260px" src="https://github.com/elrumo/macOS_Big_Sur_icons_replacements/assets/101254295/87897614-be4b-4021-8925-b8e7b8c43275">

But that also needs moderation (making sure the original work can have derivates, etc.).

**Comment by lnxfsf** [comment]
Well unless someone wants to sign up to moderate this stuff actively, I'm not sure if we can do much about this. 
Looks like dead end for now without any moderator willing to do it.
If someone wants to moderate this stuff, then reply here, so we could work on this, otherwise good feature for future.
